deepmd/train/trainer.py

```diff
@@ -539,56 +532,81 @@ def train (self,
             with open(self.profiling_file, 'w') as f:
                 f.write(chrome_trace)
 
+    def get_feed_dict (self, data, is_training) :
+        batch = data.get_batch()
+        feed_dict = {}
+        for kk in batch.keys():
+            if kk == 'find_type' or kk == 'type':
+                continue
+            if 'find_' in kk:
+                feed_dict[self.place_holders[kk]] = batch[kk]
+            else:
+                feed_dict[self.place_holders[kk]] = np.reshape(batch[kk], [-1])
+        for ii in ['type']:
+            feed_dict[self.place_holders[ii]] = np.reshape(batch[ii], [-1])
+        for ii in ['natoms_vec', 'default_mesh']:
+            feed_dict[self.place_holders[ii]] = batch[ii]
+        feed_dict[self.place_holders['is_training']] = is_training
+        return feed_dict, batch['natoms_vec']
```

Comment:
I do not understand why `batch['natoms_vec']` is returned. we can get the piece of information from `feed_dict['atoms_vec']`